src/lumigo_tracer/utils.py

```diff
@@ -259,68 +266,32 @@ def format_frame(frame_info: inspect.FrameInfo, free_space: int) -> dict:
         "lineno": frame_info.lineno,
         "fileName": frame_info.filename,
         "function": frame_info.function,
-        "variables": _truncate_locals(omit_keys(frame_info.frame.f_locals), free_space),
+        "variables": _truncate_locals(
+            frame_info.frame.f_locals, free_space, get_omitting_regexes()
+        ),
     }
 
 
-def _truncate_locals(f_locals: Dict[str, Any], free_space: int) -> FrameVariables:
+def _truncate_locals(
+    f_locals: Dict[str, Any], free_space: int, omitting_regexes: list
+) -> FrameVariables:
     """
     Truncate variable part or the entire variable in order to avoid exceeding the maximum frames size.
     :param f_locals: inspect.FrameInfo.frame.f_locals
     """
     locals_truncated: FrameVariables = {}
     for var_name, var_value in f_locals.items():
-        var = {var_name: prepare_large_data(var_value, MAX_VAR_LEN)}
+        if any(r.match(var_name) for r in omitting_regexes):  # type: ignore
+            var = {var_name: "****"}
+        else:
+            var = {var_name: lumigo_dumps(var_value, max_size=MAX_VAR_LEN)}
         free_space -= len(json.dumps(var))
         if free_space <= 0:
             return locals_truncated
         locals_truncated.update(var)
     return locals_truncated
```

Comment:
Seems like you can migrate it to use `lumigo_dumps`, you have here also scrub and json.dump
It's better to have In encapsulated service like your `lumigo_dumps`

Reply:
I wish :(
I don't have json.dump here :(

Reply:
What do you mean? `free_space -= len(json.dumps(var))`
Anyway, don't repeat here the omit logic & dump, use the dump func

Reply:
this function only dumps to calculate size - it does not really dump...
This is the problem with merging the omit and the dump - Any function that do only one thing (only omit in this case) could not use it...